zip-0402.rst

```diff
@@ -0,0 +1,166 @@
+::
+
+  ZIP: 402
+  Title: Change the fundamental wallet database from BerkeleyDB to SQLite
+  Owners: Alfredo Garcia <oxarbitrage@gmail.com>
+  Status: Draft
+  Category: Wallet
+  Created: 2020-05-30
+  License: MIT
+
+
+Terminology
+===========
+
+The key words "MUST" and "MAY" in this document are to be interpreted as described in RFC 2119. [#RFC2119]_
+
+Abstract
+========
+
+This document proposes a change of the database zcashd wallet currently use from BerkeleyDB into SQLite. It also proposes the separation of wallet transaction and other recoverable data into a separated database file.
+
+Motivation
+==========
+
+The main reason of the database change is reliability, SQLite haves a very good atomicity implementation [#SQLiteAtomic]_. Relational databases reduces redundancy and SQLite specifically is the world most used for local storage, it is also well mantained, have a good C++ interface, it is portable and open source.
+
+The database separation between recoverable versus non-recoverable data is to avoid writting too often to the file where the important non-recoverable data like public and private keys are located. Isolating transactions and others from the main wallet database will increase performance and reduce the risk of corruption.
```

Comment:
```suggestion
An advantage of the database separation between recoverable versus non-recoverable data, is to avoid writing too often to the file where the important non-recoverable data like public and private keys are located. Isolating transactions and other data that can be regenerated from the block chain from the keys in the main wallet database, will increase performance and reduce the risk of corruption.
```